### ISSUE TITLE
Incorrect "identical sub-expressions" warning for 0 and 0.0 with !== 

### ISSUE THREAD

**Issue by amcvitty** [rrrene/credo#491]
### Environment

* Credo version (`mix credo -v`): 0.8.10
* Erlang/Elixir version (`elixir -v`): Elixir 1.5.2
* Operating system: Max OS X  El Capitan 10.11.6

### What were you trying to do?

This code is trying to determine whether the number has a decimal portion

```
decimal = number - round(number)
if decimal !== 0.0 && decimal !== 0 do
   [...]
```

### Expected outcome

Removing either side of this expression results in a change in behavior, so I'd expect that credo allows this without warning. 

### Actual outcome

The code generates this warning from credo: 
```
[W] ↗ There are identical sub-expressions to the left and to the right of the
       '&&' operator.
       lib/helpers/formatting.ex:30 #(Helpers.Formatting.integer_or_float_to_duration)
```

I think it's related to the stricter equality checks === and !== . From http://elixir-lang.github.io/getting-started/basic-operators.html
>The difference between == and === is that the latter is more strict when comparing integers and >floats:

>iex> 1 == 1.0
>true
>iex> 1 === 1.0
>false

**Comment by rrrene** [comment]
@amcvitty Thanks for reporting this 😀  It is now fixed on `master`.

You can try this by setting the Credo dep to

```elixir
{:credo, github: "rrrene/credo"}
```

Please report back if your issue is solved! 👍

**Comment by amcvitty** [comment]
Look like that solved it. Thanks for the quick response!
